content/presence-occupancy/presence.textile

```diff
@@ -0,0 +1,702 @@
+  }
+});
+```
+
+```[ruby]
+members_page = channel.presence.get
+puts "#{members_page.items.length} presence members in first page"
+if members_page.has_next?
+  next_page = members_page.next
+end
+```
+
+```[python]
+members_page = channel.presence.get()
+print str(len(members_page.items)) + ' members present'
+if members_page.has_next():
+  next_page = members_page.next()
+```
+
+```[php]
+$membersPage = $channel->presence->get();
+echo(count($membersPage->items) . ' presence members in first page');
+if ($membersPage->hasNext()) {
+  $nextPage = $membersPage.next();
+}
+```
+
+```[java]
+PaginatedResult<PresenceMessage> membersPage = channel.presence.get(null);
+System.out.println(membersPage.items().length + " members in first page");
+if(membersPage.hasNext()) {
+  PaginatedResult<PresenceMessage> nextPage = membersPage.next();
+  System.out.println(nextPage.items().length + " members on 2nd page");
+}
+```
+
+```[csharp]
+  PaginatedResult<PresenceMessage> membersPage = await channel.Presence.GetAsync();
+  Console.WriteLine(membersPage.Items.Count + " members in first page");
+  if(membersPage.HasNext)
+  {
+    PaginatedResult<PresenceMessage> nextPage = await membersPage.NextAsync();
+    Console.WriteLine(nextPage.Items.Count + " members on 2nd page");
+  }
+```
+
+```[objc]
+[channel.presence get:^(ARTPaginatedResult<ARTPresenceMessage *> *membersPage, ARTErrorInfo *error) {
+  NSLog(@"%lu members in first page", [membersPage.items count]);
+  if (membersPage.hasNext) {
+    [membersPage next:^(ARTPaginatedResult<ARTPresenceMessage *> *nextPage, ARTErrorInfo *error) {
+      NSLog(@"%lu members on 2nd page", [nextPage.items count]);
+    }];
+  }
+}];
+```
+
+```[swift]
+channel.presence.get { membersPage, error in
+  let membersPage = membersPage!
+  print("\(membersPage.items.count) in first page")
+  if membersPage.hasNext {
+    membersPage.next { nextPage, error in
+      print("\(nextPage!.items.count) members on 2nd page")
+    }
+  }
+}
+```
+
+```[go]
+page, err := channel.Presence.Get(nil)
+fmt.Println("%d messages in first page\n", len(page.PresenceMessages()))
+if page.hasNext() {
+  page2, err := page.Next()
+  fmt.Println("%d messages on 2nd page!\n", len(page2.PresenceMessages()))
+}
+```
+
+h3(#batch-publish). Batch presence
+
+It is common for the presence of multiple channels to be needed. If you wish to obtain the presence of multiple channels within a single operation, you can make use of the "REST batch API":/api/rest-api#batch.
+
+h2(#presence-history). Presence History
+
+History provides access to instantaneous "live" history as well as the longer term persisted history for presence channels. If "persisted history":/storage-history/history is enabled for the channel, then presence events will "typically be stored for 24 - 72 hours":https://faqs.ably.com/how-long-are-messages-stored-for. If persisted history is not enabled, Ably retains the last two minutes of presence event history in memory.
```

Comment:
Miro suggests the what/why/when to use history should be here, but the actual how + code should be on the history page.